src/dir.rs

```diff
@@ -921,6 +921,14 @@ impl ShortNameGenerator {
         Self { short_name, chksum, name_fits, lossy_conv, basename_len: basename_len as u8, ..Default::default() }
     }
 
+    fn generate_dot() -> [u8; 11] {
+        [0x2e, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20]
```

Comment:
I don't like this repetition. Change it to the old way when you used [11; 0x20] syntax to create the array and then fill first bytes - I think compiler should optimize it to the same code. Also please add some constants for repeated padding byte. 0x20 as a short name padding is used in multiple places so file scope constant would be best. You can name it SFN_PADDING

Reply:
Here is the fixed code and the regression test (which fails without the fix)